core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala

```diff
@@ -548,6 +548,12 @@ class ColumnFormatRelation(
         throw e
     }
   }
+
+  /** Base table of this relation. */
+  override def baseTable: Option[String] = _origOptions.get("COLOCATE_WITH")
```

Comment:
I think you need the CaseInsensitive map here for lookup. See parameters in createRelation in ColumnFormatRelation.scala.